src/ol/style/expressions.js

```diff
@@ -260,24 +260,35 @@ export function expressionToGlsl(context, value, typeHint) {
     }
     return operator.toGlsl(context, value.slice(1), typeHint);
   }
+
   const valueType = getValueType(value);
   if ((valueType & ValueTypes.NUMBER) > 0) {
     return numberToGlsl(/** @type {number} */ (value));
-  } else if ((valueType & ValueTypes.BOOLEAN) > 0) {
+  }
+
+  if ((valueType & ValueTypes.BOOLEAN) > 0) {
     return value.toString();
-  } else if (
+  }
+
+  if (
     (valueType & ValueTypes.STRING) > 0 &&
     (typeHint === undefined || typeHint == ValueTypes.STRING)
   ) {
     return stringToGlsl(context, value.toString());
-  } else if (
+  }
+
+  if (
     (valueType & ValueTypes.COLOR) > 0 &&
     (typeHint === undefined || typeHint == ValueTypes.COLOR)
   ) {
     return colorToGlsl(/** @type {Array<number> | string} */ (value));
-  } else if ((valueType & ValueTypes.NUMBER_ARRAY) > 0) {
+  }
+
+  if ((valueType & ValueTypes.NUMBER_ARRAY) > 0) {
     return arrayToGlsl(/** @type {Array<number>} */ (value));
   }
+
+  throw new Error(`Unexpected expression ${value} (expected type ${typeHint})`);
```

Comment:
This is the only significant change - we now throw instead of returning `undefined`.